File: simple/python/introduce-null-object_after.py
class NullCustomer(Customer):

    def isNull(self):
        return True
    
    def getPlan(self):
        return self.NullPlan()
        
    # Some other NULL functionality.

# Replace null values with Null-object.
customer = order.customer if order.customer != None else NullCustomer()

# Use Null-object as if it's normal subclass.
plan = customer.getPlan()

File: simple/php/introduce-null-object_after.php
<?php
class NullCustomer extends Customer {
  public function isNull() {
    return true;
  }
  public function getPlan() {
    return new NullPlan();
  }
  // Some other NULL functionality.
}

// Replace null values with Null-object.
$customer = ($order->customer != null) ?
  $order->customer :
  new NullCustomer;

// Use Null-object as if it's normal subclass.
$plan = $customer->getPlan();